pvlib/shading.py

```diff
@@ -342,3 +342,144 @@ def projected_solar_zenith_angle(solar_zenith, solar_azimuth,
     # Eq. (5); angle between sun's beam and surface
     theta_T = np.degrees(np.arctan2(sx_prime, sz_prime))
     return theta_T
+
+
+def shaded_fraction1d(
+    solar_zenith,
+    solar_azimuth,
+    trackers_axis_azimuth,
+    shaded_tracker_tilt,
+    *,
+    collector_width,
+    row_pitch,
+    surface_to_axis_offset=0,
+    cross_axis_slope=0,
+    shading_tracker_tilt=None,
```

Comment:
I'm not necessarily opposed to using keyword-only parameters here, but can you explain the motivation for using them here?  And also the specific position, i.e. why are `collector_width` and `row_pitch` kw-only, but not (for example) `trackers_axis_azimuth`?

Reply:
It obviously is to avoid allowing magic numbers in user's code.

I admit what I classify as a potential magic number parameter is subjective. But my reasoning is the following:
- `solar_zenith, solar_azimuth,` are usually provided from a variable in an explicit way
- `trackers_axis_azimuth, shaded_tracker_tilt,` I think from the context it is easy to know which one is from the values.
- `collector_width, row_pitch, surface_to_axis_offset, cross_axis_slope,` these are the real candidates to be magic numbers
- `shading_tracker_tilt` in fact, I did not know what to do with this one, but given the amount of parameters in degrees, excluding this one knowing which one is any of `trackers_axis_azimuth, shaded_tracker_tilt,` seems easier.

I don't have a strong opinion other than wanting to avoid debugging unreadable code. I'm up to change this completely, I know it introduces a new way of exposing the API.

Reply:
I'd like to hear opinions regarding this 🤔

Reply:
I am tentatively in favor of using more keyword-only parameters in new functions.  I think we can always convert them to "normal" parameters later without breaking anything, so it's not a big risk in terms of API stability.  For old functions, I am not sure it is worth the deprecation and change, but perhaps it could be evaluated on a case-by-case basis.

Here are some references I found useful:
- https://docs.scipy.org/doc/scipy/dev/missing-bits.html#required-keyword-names
- https://scikit-learn-enhancement-proposals.readthedocs.io/en/latest/slep009/proposal.html

Anyway no changes needed here, I think.  Thanks for the explanation!